lib/backend/manager.go

```diff
@@ -139,3 +142,18 @@ func (m *Manager) GetClient(namespace string) (Client, error) {
 	}
 	return nil, ErrNamespaceNotFound
 }
+
+// CheckReadiness returns whether the backends are ready (by trying to download a file).
+// A backend must be explicitly configured as required for readiness to be checked.
+func (m *Manager) CheckReadiness() error {
+	for _, backend := range m.backends {
+		if !backend.mustReady {
+			continue
+		}
+		err := backend.client.Download(ReadinessCheckBlobNamespace, ReadinessCheckBlobName, bytes.NewBuffer([]byte{}))
```

Comment:
Let's not use fixed blob hashes to check for backend readiness

Reply:
Reworked.